### ISSUE TITLE
Are these permissions required?

### ISSUE THREAD

**Issue by AnastasiaLd** [microsoft/o365-moodle#1171]
We would only like to use OneDrive repository but the permissions seem a bit too broad.
In particular:

Read and write domains | 
Required to automatically detect your Office 365 tenant during setup.

This would give the application the ability to not only delete the domain from the Azure AD tenancy but create new domains as well.
As we only have 1 domain can we hardcode this in and bypass the need for the app to require the permission?

Read and write files in all site collections | 
Required for the Office 365 repository to access, download, and upload files to OneDrive.

This would give access to read/modify/delete all Sharepoint, Teams and Onedrive files.
The app does already get this access for signed in users via the delegated permission “Have full access to user files”, is this application permission required as well?

Is there a list of specific permissions by service?

Thank you

**Comment by jamesmcq** [comment]
Hi @AnastasiaLd - the permissions listed in the documentation allow the plugins to do various things, like in the descriptions you posted above. The permissions we list are the ones we test with and which allow all features of the plugins to operate correctly. You're welcome to pick and choose, omit some, or try different permissions based on your individual needs but this is not something we test for, so understand there may be glitches or problems with some combinations of permissions. I'd recommend thoroughly testing your unique permission combination before going to production. One tip - the permissions do tend to follow what you'd expect - if you only need to read OneDrive files you can probably get by without a "Write" permission, for example.

**Comment by AnastasiaLd** [comment]
Thank you James. The main issue I am experiencing is with access control links. 
I can't create access control links as I get the following error:
<img width="869" alt="Screenshot 2019-12-13 at 10 44 05" src="https://user-images.githubusercontent.com/32487197/70794582-db9c2f00-1d95-11ea-9e76-fd51ba2cc6a3.png">
I do have Microsoft OneDrive repository enabled and configured as below:
<img width="1388" alt="Screenshot 2019-12-13 at 10 47 18" src="https://user-images.githubusercontent.com/32487197/70794650-038b9280-1d96-11ea-9e82-93783b381d0c.png">
I am not sure if this is related to the permissions of the accounts or something else.
Thank you!

**Comment by porcospino** [comment]
Hello @jamesmcq 

I'm assuming you test against a disposable tenanacy.  The permissions you recommend are way too broad for production, and this is preventing us from proceeding with the integration.  We're currently working on plans for synchronous remote teaching to cope with the coronavirus.  At the moment, we are leaning towards other webinar platforms because we can't proceed on the basis of your recommendations.  Please can you advise the minimum permissions required to make this work?

**Comment by tmuras** [comment]
Hi @porcospino - I basically agree with what James wrote. Try to disable some of them and test your installation - if the features you need work, then you may be good to go.

What exactly are the permissions that you don't like and are preventing you from using the plugin?

**Comment by porcospino** [comment]
The following are too broad:
- Read and write domains (this includes delete)
- Read and write all users' full profiles (this includes delete)
- Read and write all OneNote notebooks (information disclosure)
- Read and write all groups (includes delete)
- Read and write files in all site collections (information disclosure)

The "delegated permissions" are a bit rum as well, but at least they can only trash one user's info at a time.

**Comment by nenorojas** [comment]
Hi @porcospino , @AnastasiaLd 

We performed a complete review of the Graph APIs used by the Microsoft Moodle plugins, and cleaned them up a bit, this is the link for the document with details: https://docs.google.com/spreadsheets/d/1XIFnnQ8RuFjlPGB8CiiMmK8x9Uta9XIR_icvAdx0G6k/edit#gid=0. 

We then worked with the Education team in Microsoft and created a long waited update version of the PowerShell script used to create the Azure app. The PowerShell script is included in the latest release of the plugins.

Regards,
